components/admin_console/policy_settings/policy_settings.jsx

```diff
@@ -2,63 +2,138 @@
+import BooleanSetting from '../boolean_setting.jsx';
+import ColorSetting from '../color_setting.jsx';
+import DropdownSetting from '../dropdown_setting.jsx';
+import PostEditSetting from '../post_edit_setting.jsx';
+import RadioSetting from '../radio_setting.jsx';
+import SettingsGroup from '../settings_group.jsx';
+import TextSetting from '../text_setting.jsx';
+
+import {rolesFromMapping, mappingValueFromRoles} from 'utils/policy_roles_adapter';
+
+import LoadingScreen from 'components/loading_screen.jsx';
 
 export default class PolicySettings extends AdminSettings {
+    static propTypes = {
+        actions: PropTypes.shape({
+            loadRolesIfNeeded: PropTypes.func.isRequired,
+            editRole: PropTypes.func.isRequired
+        }).isRequired
+    };
+
     constructor(props) {
         super(props);
 
         this.getConfigFromState = this.getConfigFromState.bind(this);
 
         this.renderSettings = this.renderSettings.bind(this);
+
+        this.roleBasedPolicies = {
+            allowEditPost: '',
+            restrictPostDelete: '',
+            restrictTeamInvite: '',
+            restrictPublicChannelCreation: '',
+            restrictPrivateChannelCreation: '',
+            restrictPublicChannelManagement: '',
+            restrictPrivateChannelManagement: '',
+            restrictPublicChannelDeletion: '',
+            restrictPrivateChannelDeletion: '',
+            restrictPrivateChannelManageMembers: ''
+        };
+
+        this.state = {
+            ...this.state, // Brings the state in from the parent class.
+            ...this.roleBasedPolicies,
+            loaded: false
+        };
     }
 
+    loadPoliciesIntoState(props) {
+        if (props.rolesRequest.status === RequestStatus.SUCCESS) {
+            const {roles} = props;
+
+            Object.entries(this.roleBasedPolicies).forEach(([key]) => {
+                this.roleBasedPolicies[key] = mappingValueFromRoles(key, roles);
+            });
+
+            // Adjustment to allowEditPost policy because the roles mapping is the same for 'always' and 'time_limit'
+            if (this.roleBasedPolicies.allowEditPost === Constants.ALLOW_EDIT_POST_ALWAYS && this.state.postEditTimeLimit) {
+                this.roleBasedPolicies.allowEditPost = Constants.ALLOW_EDIT_POST_TIME_LIMIT;
+            }
+
+            this.setState({...this.roleBasedPolicies, loaded: true});
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // if statement prevents over-invoking every time any props change.
+        if (nextProps.rolesRequest.status === RequestStatus.SUCCESS && this.props.rolesRequest.status !== RequestStatus.SUCCESS) {
+            this.loadPoliciesIntoState(nextProps);
+        }
+    }
+
+    componentWillMount() {
+        this.props.actions.loadRolesIfNeeded(['channel_user', 'team_user', 'channel_admin', 'team_admin', 'system_admin']).then(() => {
+            this.loadPoliciesIntoState(this.props);
```

Comment:
I'm not sure, but I think this then clause is not needed because the component will receive the status change from the redux flux, so will run this in the componentWillReceiveProps.

Reply:
@jespino Actually in my testing it is `componentWillReceiveProps` that is unnecessary.

Reply:
👍